mgmtsystem/azure.py

```diff
@@ -0,0 +1,308 @@
+# coding: utf-8
+"""Backend management system classes
+
+Used to communicate with providers without using CFME facilities
+"""
+import winrm
+import json
+from cStringIO import StringIO
+from contextlib import contextmanager
+from datetime import datetime
+from textwrap import dedent
+
+from lxml import etree
+from wait_for import wait_for
+
+from base import MgmtSystemAPIBase
+
+
+class AzureSystem(MgmtSystemAPIBase):
+    """This class is used to connect to Microsoft Azure Portal via PowerShell AzureRM Module
+    """
+    STATE_RUNNING = "VM running"
+    STATE_STOPPED = "VM deallocated"
+    STATE_STARTING = "VM starting"
+    STATE_PAUSED = "Paused"
+    STATES_STEADY = {STATE_RUNNING, STATE_PAUSED}
+    STATES_STEADY.update(STATE_STOPPED)
```

Comment:
I still it will be required going forward.  It still transistions from stopped, starting, updating, to running.  The only difference is that the PowerShell commands know this and wait to the appropriate status.  However, if we ever decide to use direct JSON calls, we will need to cycle through these.  I'll remove it for now.